### PR TITLE
Add .editorconfig and fix some indentation issues

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+indent_size = 2
+
+[*.json]
+indent_style = space
+indent_size = 2
+
+[*.md]
+trim_trailing_whitespace = false
+
+[*.php]
```
Comment (dunglas) [inline .editorconfig]: Are sections sharing the same config than * mandatory?

Comment (teohhanhui) [inline .editorconfig]: I just think it's better to make it explicit.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+indent_style = space
```
Comment (teohhanhui) [inline .editorconfig]: And it also helps show that we've made a decision for those extensions / files.

File: .editorconfig (hunk 1 above)
Comment (teohhanhui) [inline .editorconfig]: And it also helps show that we've made a decision for those extensions / files.

Comment (dunglas) [inline .editorconfig]: Fair enough.

**Comment by dunglas** [comment]
Thanks @teohhanhui

**Pull request merged**